packages/wallet-sdk/src/core/communicator/Communicator.ts

```diff
@@ -14,31 +14,24 @@ import { standardErrors } from ':core/error';
  * It also handles cleanup of event listeners and the popup window itself when necessary.
  */
 export class Communicator {
-  private url: URL;
+  private readonly url: URL;
 
   constructor(url: string = CB_KEYS_URL) {
     this.url = new URL(url);
-    this.postMessage = this.postMessage.bind(this);
-    this.onMessage = this.onMessage.bind(this);
-    this.disconnect = this.disconnect.bind(this);
   }
 
   /**
-   * Posts a message to the popup window and optionally waits for a response.
```

Comment:
`optionally waits for a response` will be handled on the SCWSigner, where we actually make RPC requests and wait for response.